pkg/controller/prometheus.go

```diff
@@ -161,20 +163,34 @@ func (c *Controller) checkAndCreatePrometheusAddlConfig(ctx context.Context, ten
 			return err
 		}
 	} else {
-		var scrapeConfigs []configmaps.ScrapeConfig
+		var scrapeConfigs, exceptedScrapeConfigs []configmaps.ScrapeConfig
 		err := yaml.Unmarshal(secret.Data[miniov2.PrometheusAddlScrapeConfigKey], &scrapeConfigs)
 		if err != nil {
 			return err
 		}
-		// Check if the scrape config is already present
-		hasScrapeConfig := false
+		// get other scrape configs
 		for _, sc := range scrapeConfigs {
-			if sc.JobName == tenant.PrometheusOperatorAddlConfigJobName() {
-				hasScrapeConfig = true
-				break
+			if !strings.HasPrefix(sc.JobName, tenant.PrometheusOperatorAddlConfigJobName()) {
+				exceptedScrapeConfigs = append(exceptedScrapeConfigs, sc)
 			}
 		}
-		if !hasScrapeConfig {
+		exceptedScrapeConfigs = append(exceptedScrapeConfigs, promCfg.ScrapeConfigs...)
```

Comment:
The only way to remove scrapeConfigs is to set `prometheusOperator` to false, then re-initialize the array. This does not seem correct.

Reply:
I guess you mean `expectedScrapeConfigs` instead of `exceptedScrapeConfigs`? Or did you mean `acceptedScapeConfigs`???